src/RESTClient.js

```diff
@@ -180,7 +180,7 @@ export class RESTClient {
      */
     static search(query, useDefaultParams, defaultParams) {
         const params = useDefaultParams ? cloneDefined(defaultParams, query) : cloneDefined(query);
-        return Object.keys(params).map(p => `${encode(p)}=${encode(params[p])}`).join('&');
+        return Object.keys(params).map(p => params[p] !== null ? `${encode(p)}=${encode(params[p])}` : '').join('&');
```

Comment:
Hmm... I think I would prefer changing the signature of the calling function to: 
```javascript
purchaseCampaignFlowUrl(campaignId, productId, voucherCode, redirectUri = this.redirectUri)
```
(adding `[]` around all of the jsdoc param args like you did for `[voucherCode]` already).

What do you think? My reason is that this feels a bit hacky to add generally to RESTClient